shs/update.sh

```
@@ -47,8 +49,14 @@ main() {
     # Handle clean install
     if [ "$CLEAN_INSTALL" -eq 1 ]; then
         if [ -d "$TARGET_DIR" ]; then
-            local backup_dir="${TARGET_DIR}_backup_$(date +%Y%m%d_%H%M%S)"
-            log_warn "Backing up to: $backup_dir"
+            # Ensure the centralized archive directory exists
+            mkdir -p "$OLD_REPOS_BASE"
+
+            # Extract folder name and set the new backup destination
+            local dir_name=$(basename "$TARGET_DIR")
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)